### PR TITLE
Enumerable challenge

### PR DESCRIPTION
**Pull request by sarazaToussaint**
## In this Pull request we:

- Create a class MyList that has an instance variable @list.
 - Implement a method #each in MyList that yields successive members of @list and uses the MyEnumerable module.
- Create a module MyEnumerable that implements the following methods (they should have the same functionality as methods in [Enumerable](https://ruby-doc.org/core-3.0.0/Enumerable.html)):
  #all?
 #any?
 #filter

### CODE REVIEW

File: module.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module MyEnumerable
```
Comment (arslanbisharat) [inline module.rb]: - [ ] Let's change the name of the file to `Enumerable.rb`.

File: class.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+require_relative 'module'
+
+class MyList
+  include MyEnumerable
+
+  def initialize(*list)
+    @list = list
+  end
+
+  # rubocop:disable Style/ExplicitBlockArgument
```
Comment (arslanbisharat) [inline class.rb]: - [ ] Let's not disable rubocop rules.

File: class.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  def each
+    @list.each { |item| yield item }
+  end
+  # rubocop:enable Style/ExplicitBlockArgument
```
Comment (arslanbisharat) [inline class.rb]: - [ ] Let's not disable rubocop rules.

File: class.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+require_relative 'module'
```
Comment (arslanbisharat) [inline class.rb]: - [ ] Please make sure `require_relative 'my_enumerable'` is present here instead of module.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @sarazaToussaint & @bamsi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @sarazaToussaint and @bamsi 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Github inline comments feature is used properly.
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**